src/linker/Linker/MethodReferenceExtensions.cs

```diff
@@ -11,7 +11,9 @@ public static class MethodReferenceExtensions
 		public static string GetDisplayName (this MethodReference method)
 		{
 			var builder = new System.Text.StringBuilder ();
-			builder.Append (method.DeclaringType.FullName);
+			builder.Append ((method as MemberReference).GetNamespaceDisplayName ());
```

Comment:
Why do you need the "as" here - MethodReference should always be a MemberReference.